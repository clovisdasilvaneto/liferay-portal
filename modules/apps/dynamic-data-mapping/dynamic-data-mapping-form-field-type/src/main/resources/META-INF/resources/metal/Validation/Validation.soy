{namespace Validation}

{template .render}
	{@param name: string}
	{@param? id: string}
	{@param? spritemap: string}
	{@param? dataType: string}
	{@param? enableValidation: bool}
	{@param? errorMessage: string}
	{@param? label: string}
	{@param? parameterMessage: string}
	{@param? strings: [chooseAnOption: string, email: string, errorMessage: string, ifInput: string, showErrorMessage: string, theValue: string, url: string, validationMessage: string]}
	{@param? _handleEnableValidationEdited: any}
	{@param? _handleValidationValueEdited: any}
	{@param? _handleParameterMessageEdited: any}
	{@param? _handleErrorMessageEdited: any}
	{@param? selectedValidation: [parameterMessage: string, value: string]}
	{@param? validations: list<[
		checked: bool,
		disabled: bool,
		id: string,
		inline: bool,
		label: string,
		name: string,
		showLabel: bool,
		value: string
	]>}

	{call .content}
		{param _handleEnableValidationEdited: $_handleEnableValidationEdited /}
		{param _handleErrorMessageEdited: $_handleErrorMessageEdited /}
		{param _handleParameterMessageEdited: $_handleParameterMessageEdited /}
		{param dataType: $dataType /}
		{param enableValidation: $enableValidation /}
		{param errorMessage: $errorMessage /}
		{param _handleValidationValueEdited: $_handleValidationValueEdited /}
		{param id: $id /}
		{param label: $label /}
		{param name: $name /}
		{param parameterMessage: $parameterMessage /}
		{param selectedValidation: $selectedValidation /}
		{param spritemap: $spritemap /}
		{param strings: $strings /}
		{param validations: $validations /}
	{/call}

{/template}

{template .content}
	{@param name: string}
	{@param? _handleEnableValidationEdited: any}
	{@param? _handleErrorMessageEdited: any}
	{@param? _handleParameterMessageEdited: any}
	{@param? _handleValidationValueEdited: any}
	{@param? dataType: string}
	{@param? enableValidation: bool}
	{@param? errorMessage: string}
	{@param? label: string}
	{@param? parameterMessage: string}
	{@param? selectedValidation: [parameterMessage: string, value: string]}
	{@param? spritemap: string}
	{@param? strings: [chooseAnOption: string, email: string, errorMessage: string, ifInput: string, showErrorMessage: string, theValue: string, url: string, validationMessage: string]}
	{@param? validations: list<[
		checked: bool,
		disabled: bool,
		id: string,
		inline: bool,
		label: string,
		name: string,
		showLabel: bool,
		value: string
	]>}

	<div class="form-group lfr-ddm-form-field-validation">
		{call Checkbox.render}
			{param events: [
				'fieldEdited': $_handleEnableValidationEdited
			] /}
			{param label: $label /}
			{param ref: 'enableValidation' /}
			{param value: $enableValidation /}
			{param showAsSwitcher: true /}
			{param spritemap: $spritemap /}
		{/call}

		{call Select.render}
			{param disableEmptyOption: true /}
			{param events: [
				'fieldEdited': $_handleValidationValueEdited
			] /}
			{param label: $strings.ifInput /}
			{param options: $validations /}
			{param placeholder: $strings?.chooseAnOption /}
			{param ref: 'selectedValidation' /}
			{param spritemap: $spritemap /}
			{param strings: ['chooseAnOption': $strings?.chooseAnOption] /}
			{param value: $selectedValidation?.value /}
			{param visible: $enableValidation /}
		{/call}

		{if $selectedValidation?.parameterMessage and $dataType == 'string'}
			{call Text.render}
				{param dataType: $dataType /}
				{param events: [
					'fieldEdited': $_handleParameterMessageEdited
				] /}
				{param label: $strings?.theValue /}
				{param name: $name + '_parameterMessage' /}
				{param placeholder: $selectedValidation?.parameterMessage /}
				{param ref: 'parameterMessage' /}
				{param required: true /}
				{param spritemap: $spritemap /}
				{param value: $parameterMessage /}
				{param visible: $enableValidation /}
			{/call}
		{elseif $selectedValidation?.parameterMessage}
			{call Numeric.render}
				{param dataType: $dataType /}
				{param events: [
					'fieldEdited': $_handleParameterMessageEdited
				] /}
				{param label: $strings?.theValue /}
				{param name: $name + '_parameterMessage' /}
				{param placeholder: $selectedValidation?.parameterMessage /}
				{param ref: 'parameterMessage' /}
				{param required: true /}
				{param spritemap: $spritemap /}
				{param value: $parameterMessage /}
				{param visible: $enableValidation /}
			{/call}
		{/if}

		{call Text.render}
			{param events: [
				'fieldEdited': $_handleErrorMessageEdited
			] /}
			{param label: $strings?.showErrorMessage /}
			{param name: $name + '_errorMessage' /}
			{param placeholder: $strings?.showErrorMessage /}
			{param ref: 'errorMessage' /}
			{param required: true /}
			{param spritemap: $spritemap /}
			{param value: $errorMessage /}
			{param visible: $enableValidation /}
		{/call}
	</div>
{/template}