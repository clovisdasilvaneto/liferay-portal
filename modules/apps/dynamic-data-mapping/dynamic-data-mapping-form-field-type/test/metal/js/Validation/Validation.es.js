import Validation from 'source/Validation/Validation.es';
import {dom as MetalTestUtil} from 'metal-dom';

let component;
const spritemap = 'icons.svg';

describe(
	'Validation',
	() => {
		afterEach(
			() => {
				if (component) {
					component.dispose();
				}
			}
		);

		it(
			'should render checkbox to enable Validation',
			() => {
				component = new Validation(
					{
						dataType: 'string',
						label: 'Validator',
						name: 'validation',
						spritemap
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should enable validation after click on toogle',
			() => {
				jest.useFakeTimers();

				component = new Validation(
					{
						dataType: 'string',
						label: 'Validator',
						name: 'validation',
						spritemap
					}
				);

				MetalTestUtil.triggerEvent(
					component.element.querySelector('input[type=checkbox]'),
					'change',
					{}
				);

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should render parameter field with TextField element',
			() => {
				jest.useFakeTimers();

				component = new Validation(
					{
						dataType: 'string',
						enableValidation: true,
						label: 'Validator',
						name: 'validation',
						spritemap
					}
				);

				component.handleValidationValue({value: ['notContains']});

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should render parameter field with Numeric element',
			() => {
				jest.useFakeTimers();

				component = new Validation(
					{
						dataType: 'numeric',
						enableValidation: true,
						label: 'Validator',
						name: 'validation',
						spritemap
					}
				);

				component.handleValidationValue({value: ['lt']});

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should render parameter field with Text element and then with Numeric after update dataType',
			() => {
				jest.useFakeTimers();

				component = new Validation(
					{
						dataType: 'string',
						enableValidation: true,
						label: 'Validator',
						name: 'validation',
						spritemap,
						validation: {
							dataType: undefined
						}
					}
				);

				component.handleValidationValue({value: ['notContains']});

				jest.runAllTimers();

				expect(component).toMatchSnapshot();

				component.dataType = 'numeric';

				jest.runAllTimers();

				component.handleValidationValue({value: ['lt']});

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should not render parameter field if validation is isUrl',
			() => {
				jest.useFakeTimers();

				component = new Validation(
					{
						dataType: 'string',
						enableValidation: true,
						label: 'Validator',
						name: 'validation',
						spritemap
					}
				);

				component.handleValidationValue({value: ['url']});

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should not render parameter field if validation is isEmailAddress',
			() => {
				jest.useFakeTimers();

				component = new Validation(
					{
						dataType: 'string',
						enableValidation: true,
						label: 'Validator',
						name: 'validation',
						spritemap
					}
				);

				component.handleValidationValue({value: ['email']});

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should render filled fields when value is filled',
			() => {
				jest.useFakeTimers();

				component = new Validation(
					{
						dataType: 'string',
						label: 'Validator',
						name: 'validation',
						spritemap,
						value: {
							errorMessage: 'Error message',
							expression: 'NOT(contains(textfield, "forbiddenWord"))'
						}
					}
				);

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should update parameter message and emit a field edited event',
			() => {
				jest.useFakeTimers();

				const handleFieldEdited = jest.fn();

				const events = {fieldEdited: handleFieldEdited};

				component = new Validation(
					{
						dataType: 'string',
						enableValidation: true,
						events,
						label: 'Validator',
						name: 'validation',
						spritemap,
						validation: {
							fieldName: 'textfield'
						},
						value: {
							errorMessage: 'Error message',
							expression: 'NOT(contains(textfield, "forbiddenWord"))'
						}
					}
				);

				jest.runAllTimers();

				const parameterComponent = component.element.querySelector(
					'input[name="validation_parameterMessage"]'
				);

				parameterComponent.value = 'new Parameter Message';

				MetalTestUtil.triggerEvent(parameterComponent, 'input', {});

				jest.runAllTimers();

				expect(handleFieldEdited).toHaveBeenCalled();
				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should update error message and emit a field edited event',
			() => {
				jest.useFakeTimers();

				const handleFieldEdited = jest.fn();

				const events = {fieldEdited: handleFieldEdited};

				component = new Validation(
					{
						dataType: 'string',
						enableValidation: true,
						events,
						label: 'Validator',
						name: 'validation',
						spritemap,
						validation: {
							fieldName: 'textfield'
						},
						value: {
							errorMessage: 'Error message',
							expression: 'NOT(contains(textfield, "forbiddenWord"))'
						}
					}
				);

				jest.runAllTimers();

				const errorComponent = component.element.querySelector(
					'input[name="validation_errorMessage"]'
				);

				errorComponent.value = 'new Error Message';

				MetalTestUtil.triggerEvent(errorComponent, 'input', {});

				jest.runAllTimers();

				expect(handleFieldEdited).toHaveBeenCalled();
				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should propagate the field edit event',
			() => {
				jest.useFakeTimers();

				component = new Validation(
					{
						dataType: 'string',
						enableValidation: true,
						name: 'validation',
						spritemap,
						validation: {
							fieldName: 'textfield'
						},
						value: {
							errorMessage: 'Error message',
							expression: 'NOT(contains(textfield, "forbiddenWord"))'
						}
					}
				);

				const spy = jest.spyOn(component, 'emit');

				jest.runAllTimers();

				const errorComponent = component.element.querySelector(
					'input[name="validation_errorMessage"]'
				);

				errorComponent.value = 'new Error Message';

				MetalTestUtil.triggerEvent(errorComponent, 'input', {});

				jest.runAllTimers();

				expect(spy).toHaveBeenCalled();
				expect(spy).toHaveBeenCalledWith('fieldEdited', expect.any(Object));
			}
		);
	}
);