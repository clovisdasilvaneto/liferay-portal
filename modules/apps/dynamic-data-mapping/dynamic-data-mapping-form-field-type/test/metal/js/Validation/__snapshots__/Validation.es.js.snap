// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validation should enable validation after click on toogle 1`] = `
<div class="form-group lfr-ddm-form-field-validation">
  <div class="form-group">
    <label class="ddm-toggle-switch toggle-switch">
      <input class="toggle-switch-check" type="checkbox">
      <span aria-hidden="true" class="toggle-switch-bar">
        <span class="toggle-switch-handle"></span>
        <span class="toggle-switch-text toggle-switch-text-right">Validator</span>
      </span>
    </label>
  </div>
  <div class="form-group" style="display: none;">
    <label>if-input </label>
    <div class="form-builder-select-field input-group-container">
      <div class="form-control select-field-trigger">
        <div class="option-selected option-selected-placeholder">choose-an-option</div>
        <a class="select-arrow-down-container" href="javascript:;">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-double" focusable="false">
            <title>caret-double</title>
            <use xlink:href="icons.svg#caret-double"></use>
          </svg>
        </a>
      </div>
    </div>
    <div class="dropdown-menu">
      <ul class="list-unstyled">
        <li class="dropdown-item" data-option-value="">choose-an-option</li>
        <li class="dropdown-item" data-option-value="notContains">contains</li>
        <li class="dropdown-item" data-option-value="contains">does-not-contain</li>
        <li class="dropdown-item" data-option-value="url">is-not-url</li>
        <li class="dropdown-item" data-option-value="email">is-not-email</li>
        <li class="dropdown-item" data-option-value="regularExpression">does-not-match</li>
      </ul>
    </div>
  </div>
  <div class="form-group" style="display: none;">
    <label>show-error-message
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_errorMessage" placeholder="show-error-message" type="text" value="">
  </div>
</div>
`;

exports[`Validation should not render parameter field if validation is isEmailAddress 1`] = `
<div class="form-group lfr-ddm-form-field-validation">
  <div class="form-group">
    <label class="ddm-toggle-switch toggle-switch">
      <input checked="" class="toggle-switch-check" type="checkbox">
      <span aria-hidden="true" class="toggle-switch-bar">
        <span class="toggle-switch-handle"></span>
        <span class="toggle-switch-text toggle-switch-text-right">Validator</span>
      </span>
    </label>
  </div>
  <div class="form-group">
    <label>if-input </label>
    <div class="form-builder-select-field input-group-container">
      <div class="form-control select-field-trigger">
        <div class="option-selected option-selected-placeholder">is-not-email</div>
        <a class="select-arrow-down-container" href="javascript:;">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-double" focusable="false">
            <title>caret-double</title>
            <use xlink:href="icons.svg#caret-double"></use>
          </svg>
        </a>
      </div>
    </div>
    <div class="dropdown-menu">
      <ul class="list-unstyled">
        <li class="dropdown-item" data-option-value="">choose-an-option</li>
        <li class="dropdown-item" data-option-value="notContains">contains</li>
        <li class="dropdown-item" data-option-value="contains">does-not-contain</li>
        <li class="dropdown-item" data-option-value="url">is-not-url</li>
        <li class="dropdown-item" data-option-value="email">is-not-email</li>
        <li class="dropdown-item" data-option-value="regularExpression">does-not-match</li>
      </ul>
    </div>
  </div>
  <div class="form-group">
    <label>show-error-message
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_errorMessage" placeholder="show-error-message" type="text" value="">
  </div>
</div>
`;

exports[`Validation should not render parameter field if validation is isUrl 1`] = `
<div class="form-group lfr-ddm-form-field-validation">
  <div class="form-group">
    <label class="ddm-toggle-switch toggle-switch">
      <input checked="" class="toggle-switch-check" type="checkbox">
      <span aria-hidden="true" class="toggle-switch-bar">
        <span class="toggle-switch-handle"></span>
        <span class="toggle-switch-text toggle-switch-text-right">Validator</span>
      </span>
    </label>
  </div>
  <div class="form-group">
    <label>if-input </label>
    <div class="form-builder-select-field input-group-container">
      <div class="form-control select-field-trigger">
        <div class="option-selected option-selected-placeholder">is-not-url</div>
        <a class="select-arrow-down-container" href="javascript:;">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-double" focusable="false">
            <title>caret-double</title>
            <use xlink:href="icons.svg#caret-double"></use>
          </svg>
        </a>
      </div>
    </div>
    <div class="dropdown-menu">
      <ul class="list-unstyled">
        <li class="dropdown-item" data-option-value="">choose-an-option</li>
        <li class="dropdown-item" data-option-value="notContains">contains</li>
        <li class="dropdown-item" data-option-value="contains">does-not-contain</li>
        <li class="dropdown-item" data-option-value="url">is-not-url</li>
        <li class="dropdown-item" data-option-value="email">is-not-email</li>
        <li class="dropdown-item" data-option-value="regularExpression">does-not-match</li>
      </ul>
    </div>
  </div>
  <div class="form-group">
    <label>show-error-message
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_errorMessage" placeholder="show-error-message" type="text" value="">
  </div>
</div>
`;

exports[`Validation should render checkbox to enable Validation 1`] = `
<div class="form-group lfr-ddm-form-field-validation">
  <div class="form-group">
    <label class="ddm-toggle-switch toggle-switch">
      <input class="toggle-switch-check" type="checkbox">
      <span aria-hidden="true" class="toggle-switch-bar">
        <span class="toggle-switch-handle"></span>
        <span class="toggle-switch-text toggle-switch-text-right">Validator</span>
      </span>
    </label>
  </div>
  <div class="form-group" style="display: none;">
    <label>if-input </label>
    <div class="form-builder-select-field input-group-container">
      <div class="form-control select-field-trigger">
        <div class="option-selected option-selected-placeholder">choose-an-option</div>
        <a class="select-arrow-down-container" href="javascript:;">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-double" focusable="false">
            <title>caret-double</title>
            <use xlink:href="icons.svg#caret-double"></use>
          </svg>
        </a>
      </div>
    </div>
    <div class="dropdown-menu">
      <ul class="list-unstyled">
        <li class="dropdown-item" data-option-value="">choose-an-option</li>
        <li class="dropdown-item" data-option-value="notContains">contains</li>
        <li class="dropdown-item" data-option-value="contains">does-not-contain</li>
        <li class="dropdown-item" data-option-value="url">is-not-url</li>
        <li class="dropdown-item" data-option-value="email">is-not-email</li>
        <li class="dropdown-item" data-option-value="regularExpression">does-not-match</li>
      </ul>
    </div>
  </div>
  <div class="form-group" style="display: none;">
    <label>show-error-message
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_errorMessage" placeholder="show-error-message" type="text" value="">
  </div>
</div>
`;

exports[`Validation should render filled fields when value is filled 1`] = `
<div class="form-group lfr-ddm-form-field-validation">
  <div class="form-group">
    <label class="ddm-toggle-switch toggle-switch">
      <input checked="" class="toggle-switch-check" type="checkbox">
      <span aria-hidden="true" class="toggle-switch-bar">
        <span class="toggle-switch-handle"></span>
        <span class="toggle-switch-text toggle-switch-text-right">Validator</span>
      </span>
    </label>
  </div>
  <div class="form-group">
    <label>if-input </label>
    <div class="form-builder-select-field input-group-container">
      <div class="form-control select-field-trigger">
        <div class="option-selected option-selected-placeholder">contains</div>
        <a class="select-arrow-down-container" href="javascript:;">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-double" focusable="false">
            <title>caret-double</title>
            <use xlink:href="icons.svg#caret-double"></use>
          </svg>
        </a>
      </div>
    </div>
    <div class="dropdown-menu">
      <ul class="list-unstyled">
        <li class="dropdown-item" data-option-value="">choose-an-option</li>
        <li class="dropdown-item" data-option-value="notContains">contains</li>
        <li class="dropdown-item" data-option-value="contains">does-not-contain</li>
        <li class="dropdown-item" data-option-value="url">is-not-url</li>
        <li class="dropdown-item" data-option-value="email">is-not-email</li>
        <li class="dropdown-item" data-option-value="regularExpression">does-not-match</li>
      </ul>
    </div>
  </div>
  <div class="form-group">
    <label>the-value
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_parameterMessage" placeholder="text" type="text" value="forbiddenWord">
  </div>
  <div class="form-group">
    <label>show-error-message
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_errorMessage" placeholder="show-error-message" type="text" value="Error message">
  </div>
</div>
`;

exports[`Validation should render parameter field with Numeric element 1`] = `
<div class="form-group lfr-ddm-form-field-validation">
  <div class="form-group">
    <label class="ddm-toggle-switch toggle-switch">
      <input checked="" class="toggle-switch-check" type="checkbox">
      <span aria-hidden="true" class="toggle-switch-bar">
        <span class="toggle-switch-handle"></span>
        <span class="toggle-switch-text toggle-switch-text-right">Validator</span>
      </span>
    </label>
  </div>
  <div class="form-group">
    <label>if-input </label>
    <div class="form-builder-select-field input-group-container">
      <div class="form-control select-field-trigger">
        <div class="option-selected option-selected-placeholder">is-greater-than-or-equal-to</div>
        <a class="select-arrow-down-container" href="javascript:;">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-double" focusable="false">
            <title>caret-double</title>
            <use xlink:href="icons.svg#caret-double"></use>
          </svg>
        </a>
      </div>
    </div>
    <div class="dropdown-menu">
      <ul class="list-unstyled">
        <li class="dropdown-item" data-option-value="">choose-an-option</li>
        <li class="dropdown-item" data-option-value="lt">is-greater-than-or-equal-to</li>
        <li class="dropdown-item" data-option-value="lteq">is-greater-than</li>
        <li class="dropdown-item" data-option-value="eq">is-not-equal-to</li>
        <li class="dropdown-item" data-option-value="gt">is-less-than-or-equal-to</li>
        <li class="dropdown-item" data-option-value="gteq">is-less-than</li>
      </ul>
    </div>
  </div>
  <div class="form-group">
    <label>show-error-message
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_errorMessage" placeholder="show-error-message" type="text" value="">
  </div>
</div>
`;

exports[`Validation should render parameter field with Text element and then with Numeric after update dataType 1`] = `
<div class="form-group lfr-ddm-form-field-validation">
  <div class="form-group">
    <label class="ddm-toggle-switch toggle-switch">
      <input checked="" class="toggle-switch-check" type="checkbox">
      <span aria-hidden="true" class="toggle-switch-bar">
        <span class="toggle-switch-handle"></span>
        <span class="toggle-switch-text toggle-switch-text-right">Validator</span>
      </span>
    </label>
  </div>
  <div class="form-group">
    <label>if-input </label>
    <div class="form-builder-select-field input-group-container">
      <div class="form-control select-field-trigger">
        <div class="option-selected option-selected-placeholder">contains</div>
        <a class="select-arrow-down-container" href="javascript:;">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-double" focusable="false">
            <title>caret-double</title>
            <use xlink:href="icons.svg#caret-double"></use>
          </svg>
        </a>
      </div>
    </div>
    <div class="dropdown-menu">
      <ul class="list-unstyled">
        <li class="dropdown-item" data-option-value="">choose-an-option</li>
        <li class="dropdown-item" data-option-value="notContains">contains</li>
        <li class="dropdown-item" data-option-value="contains">does-not-contain</li>
        <li class="dropdown-item" data-option-value="url">is-not-url</li>
        <li class="dropdown-item" data-option-value="email">is-not-email</li>
        <li class="dropdown-item" data-option-value="regularExpression">does-not-match</li>
      </ul>
    </div>
  </div>
  <div class="form-group">
    <label>show-error-message
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_errorMessage" placeholder="show-error-message" type="text" value="">
  </div>
</div>
`;

exports[`Validation should render parameter field with Text element and then with Numeric after update dataType 2`] = `
<div class="form-group lfr-ddm-form-field-validation">
  <div class="form-group">
    <label class="ddm-toggle-switch toggle-switch">
      <input checked="" class="toggle-switch-check" type="checkbox">
      <span aria-hidden="true" class="toggle-switch-bar">
        <span class="toggle-switch-handle"></span>
        <span class="toggle-switch-text toggle-switch-text-right">Validator</span>
      </span>
    </label>
  </div>
  <div class="form-group">
    <label>if-input </label>
    <div class="form-builder-select-field input-group-container">
      <div class="form-control select-field-trigger">
        <div class="option-selected option-selected-placeholder">is-greater-than-or-equal-to</div>
        <a class="select-arrow-down-container" href="javascript:;">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-double" focusable="false">
            <title>caret-double</title>
            <use xlink:href="icons.svg#caret-double"></use>
          </svg>
        </a>
      </div>
    </div>
    <div class="dropdown-menu">
      <ul class="list-unstyled">
        <li class="dropdown-item" data-option-value="">choose-an-option</li>
        <li class="dropdown-item" data-option-value="lt">is-greater-than-or-equal-to</li>
        <li class="dropdown-item" data-option-value="lteq">is-greater-than</li>
        <li class="dropdown-item" data-option-value="eq">is-not-equal-to</li>
        <li class="dropdown-item" data-option-value="gt">is-less-than-or-equal-to</li>
        <li class="dropdown-item" data-option-value="gteq">is-less-than</li>
      </ul>
    </div>
  </div>
  <div class="form-group">
    <label>show-error-message
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_errorMessage" placeholder="show-error-message" type="text" value="">
  </div>
</div>
`;

exports[`Validation should render parameter field with TextField element 1`] = `
<div class="form-group lfr-ddm-form-field-validation">
  <div class="form-group">
    <label class="ddm-toggle-switch toggle-switch">
      <input checked="" class="toggle-switch-check" type="checkbox">
      <span aria-hidden="true" class="toggle-switch-bar">
        <span class="toggle-switch-handle"></span>
        <span class="toggle-switch-text toggle-switch-text-right">Validator</span>
      </span>
    </label>
  </div>
  <div class="form-group">
    <label>if-input </label>
    <div class="form-builder-select-field input-group-container">
      <div class="form-control select-field-trigger">
        <div class="option-selected option-selected-placeholder">contains</div>
        <a class="select-arrow-down-container" href="javascript:;">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-double" focusable="false">
            <title>caret-double</title>
            <use xlink:href="icons.svg#caret-double"></use>
          </svg>
        </a>
      </div>
    </div>
    <div class="dropdown-menu">
      <ul class="list-unstyled">
        <li class="dropdown-item" data-option-value="">choose-an-option</li>
        <li class="dropdown-item" data-option-value="notContains">contains</li>
        <li class="dropdown-item" data-option-value="contains">does-not-contain</li>
        <li class="dropdown-item" data-option-value="url">is-not-url</li>
        <li class="dropdown-item" data-option-value="email">is-not-email</li>
        <li class="dropdown-item" data-option-value="regularExpression">does-not-match</li>
      </ul>
    </div>
  </div>
  <div class="form-group">
    <label>show-error-message
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_errorMessage" placeholder="show-error-message" type="text" value="">
  </div>
</div>
`;

exports[`Validation should update error message and emit a field edited event 1`] = `
<div class="form-group lfr-ddm-form-field-validation">
  <div class="form-group">
    <label class="ddm-toggle-switch toggle-switch">
      <input checked="" class="toggle-switch-check" type="checkbox">
      <span aria-hidden="true" class="toggle-switch-bar">
        <span class="toggle-switch-handle"></span>
        <span class="toggle-switch-text toggle-switch-text-right">Validator</span>
      </span>
    </label>
  </div>
  <div class="form-group">
    <label>if-input </label>
    <div class="form-builder-select-field input-group-container">
      <div class="form-control select-field-trigger">
        <div class="option-selected option-selected-placeholder">contains</div>
        <a class="select-arrow-down-container" href="javascript:;">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-double" focusable="false">
            <title>caret-double</title>
            <use xlink:href="icons.svg#caret-double"></use>
          </svg>
        </a>
      </div>
    </div>
    <div class="dropdown-menu">
      <ul class="list-unstyled">
        <li class="dropdown-item" data-option-value="">choose-an-option</li>
        <li class="dropdown-item" data-option-value="notContains">contains</li>
        <li class="dropdown-item" data-option-value="contains">does-not-contain</li>
        <li class="dropdown-item" data-option-value="url">is-not-url</li>
        <li class="dropdown-item" data-option-value="email">is-not-email</li>
        <li class="dropdown-item" data-option-value="regularExpression">does-not-match</li>
      </ul>
    </div>
  </div>
  <div class="form-group">
    <label>the-value
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_parameterMessage" placeholder="text" type="text" value="forbiddenWord">
  </div>
  <div class="form-group">
    <label>show-error-message
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_errorMessage" placeholder="show-error-message" type="text" value="new Error Message">
  </div>
</div>
`;

exports[`Validation should update parameter message and emit a field edited event 1`] = `
<div class="form-group lfr-ddm-form-field-validation">
  <div class="form-group">
    <label class="ddm-toggle-switch toggle-switch">
      <input checked="" class="toggle-switch-check" type="checkbox">
      <span aria-hidden="true" class="toggle-switch-bar">
        <span class="toggle-switch-handle"></span>
        <span class="toggle-switch-text toggle-switch-text-right">Validator</span>
      </span>
    </label>
  </div>
  <div class="form-group">
    <label>if-input </label>
    <div class="form-builder-select-field input-group-container">
      <div class="form-control select-field-trigger">
        <div class="option-selected option-selected-placeholder">contains</div>
        <a class="select-arrow-down-container" href="javascript:;">
          <svg aria-hidden="true" class="lexicon-icon lexicon-icon-caret-double" focusable="false">
            <title>caret-double</title>
            <use xlink:href="icons.svg#caret-double"></use>
          </svg>
        </a>
      </div>
    </div>
    <div class="dropdown-menu">
      <ul class="list-unstyled">
        <li class="dropdown-item" data-option-value="">choose-an-option</li>
        <li class="dropdown-item" data-option-value="notContains">contains</li>
        <li class="dropdown-item" data-option-value="contains">does-not-contain</li>
        <li class="dropdown-item" data-option-value="url">is-not-url</li>
        <li class="dropdown-item" data-option-value="email">is-not-email</li>
        <li class="dropdown-item" data-option-value="regularExpression">does-not-match</li>
      </ul>
    </div>
  </div>
  <div class="form-group">
    <label>the-value
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_parameterMessage" placeholder="text" type="text" value="new Parameter Message">
  </div>
  <div class="form-group">
    <label>show-error-message
      <span class="reference-mark">
        <svg aria-hidden="true" class="lexicon-icon lexicon-icon-asterisk" focusable="false">
          <title>asterisk</title>
          <use xlink:href="icons.svg#asterisk"></use>
        </svg>
      </span>
    </label>
    <input class="form-control" name="validation_errorMessage" placeholder="show-error-message" type="text" value="Error message">
  </div>
</div>
`;
